autorel/dist_tarball_builder/dist_tarball.py

```diff
@@ -0,0 +1,158 @@
+"""
+    module : dist_tarball
+    This module is responsible for building syslog-ng from source
+    in a docker environment and providing functionality of exporting
+    the generated distribution tarball.
+"""
+import logging
+import tempfile
+import pygit2
+import os
+import sys
+import glob
+from docker_cli import Docker
+
+
+
+class DistTarballBuilder(object):
+    """
+        Responsible for downloading souce code and carry out
+        build process to generate the distribution tarball.
+    """
+    def __init__(self, project, source_url, image_name):
+        self._project = project
+        self._source_url = source_url
+        self._image_name = image_name
+        self._docker_cli = Docker()
+        # configure logger
+        self._logger = logging.getLogger(__name__)
+        self._logger.setLevel(logging.DEBUG)
+        channel = logging.StreamHandler(sys.stdout)
+        channel.setLevel(logging.DEBUG)
+        formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+        channel.setFormatter(formatter)
+        self._logger.addHandler(channel)
+
+
+    def _download_source(self):
+        """
+            Downloads the source code at a temporary directory
+            on the host system
+        """
+        self._source_directory = tempfile.mkdtemp()
+        repo_path = os.path.join(self._source_directory,
+                                 'syslog-ng')
+        pygit2.clone_repository(self._source_url,
```

Comment:
Hmm... the repository should be cloned _before_ DistTarballBuilder come into play...